Rename test definitions

TaekyungHeo opened the pull request:
## Summary
Naming conventions in CloudAI are not accurate. UCC and NCCL are libraries, while UCC tests and NCCL tests are what we are running in CloudAI. Chakra is a project that includes Chakra replay as a tool.

* Rename UCCCmdArgs to UCCTestCmdArgs
* Rename NCCLCmdArgs to NCCLTestCmdArgs
* Rename Chakra to ChakraReplay

## Test Plan
CI passes

amaslenn reviewed (commented):
I agree with Chakra change, but `UCCCmdArgs` -> `UCCTestCmdArgs` seems unreasonable. It is also longer now. And following this logic, `UCCTestDefinition` should become `UCCTestTestDefinition` now.

P.S. all our "tests" are better described as "workloads" IMO:
1. `UCCTestDefinition` -> `UCCWorkload`
2. `TestScenario` -> `Suite`. `Tests` inside scenario -> `Cases`.
3. etc.

TaekyungHeo commented:
@srinivas212 & @srivatsankrishnan, please review our discussion. Let's talk about this later.

I incorporated @amaslenn's comments and reverted the changes to the UCC and NCCL test names. Let's merge what we've already agreed on first.
